src/lambda/commands/deploySamApplication.ts

```diff
@@ -30,40 +31,71 @@ export async function deploySamApplication({
     }
 
     const { template, s3Bucket, stackName } = args
+    const deployApplicationPromise = new Promise<void>(async (resolve, reject) => {
```

Comment:
I don't think we need to manually create a promise here, just use an `async (): Promise<void>` method instead.

Reply:
I'd prefer not to put that directly into the `setStatusBarMessage` call, it feels harder to read.

Reply:
You don't need to, just replace:

```typescript
const deployApplicationPromise = new Promise<void>(async (resolve, reject) => {
    // ....
})
```

with

```typescript
const deployApplicationPromise = async () => {
    // ...
}()
```

Up to you though.